### PR TITLE
Sanjo/channel routing fixes

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

**Comment by SanjoDeundiak** [comment]
Changes by @spacekookie that make Channels feasible

**Review by spacekookie — APPROVED** [review]
Looks good!

File: implementations/rust/ockam/ockam_core/src/message.rs [hunk 1]
```diff
@@ -43,25 +43,36 @@ impl From<serde_bare::Error> for crate::Error {
 /// without requiring changes in the user message types.
 pub struct Routed<M: Message> {
     inner: M,
-    route: Route,
+    return_: Route,
+    onward: Route,
```
Comment (mrinalwadhwa) [inline implementations/rust/ockam/ockam_core/src/message.rs:47]: could we make these fields `return_route` and `onward_route`?

Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_core/src/message.rs:47]: I think we should do that in a separate PR. @spacekookie what do you think?

Comment (mrinalwadhwa) [reply implementations/rust/ockam/ockam_core/src/message.rs:47]: Different PR works 👍

Comment (spacekookie) [reply implementations/rust/ockam/ockam_core/src/message.rs:47]: Sorry, I didn't see the followup messages before doing a push. I reverted the commit. Will open a PR in a sec

**Comment by jared-s** [comment]
Merged https://github.com/ockam-network/ockam/pull/1152

**Pull request closed without merging**